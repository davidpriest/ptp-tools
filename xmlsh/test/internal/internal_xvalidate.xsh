# test of xinclude

# Basic test
echo Validating with DTD
xvalidate  -dtd ../../samples/data/books.dtd < ../../samples/data/books.xml || echo Validation failed

echo Validating with xsd
xvalidate  -xsd ../../samples/data/books.xsd  ../../samples/data/books.xml || echo Validation failed


echo validate  -rng ../../samples/data/books.rng  ../../samples/data/books.xml || echo Validation failed

echo Following should fail
xvalidate  -xsd ../../samples/data/books.xsd  ../../samples/data/othello.xml  2>/dev/null || echo Successfully trapped invalid schema

echo Schematron validation
N=$(schematron ../../samples/data/books.sch ../../samples/data/books.xml | xmlns:svrl=http://purl.oclc.org/dsdl/svrl xpath 'count(//svrl:fired-rule)')
echo Matching Items: $N

N=$(xvalidate -schematron ../../samples/data/books.sch ../../samples/data/books.xml | xmlns:svrl=http://purl.oclc.org/dsdl/svrl xpath 'count(//svrl:fired-rule)')
echo Matching Items: $N

echo RNG Validation
xvalidate -rng ../../samples/data/books.rng ../../samples/data/books.xml || { echo failed RNG validation ; exit 1 ; }



