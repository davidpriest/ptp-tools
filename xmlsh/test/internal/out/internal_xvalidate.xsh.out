Validating with DTD
Validating with xsd
validate -rng ../../samples/data/books.rng ../../samples/data/books.xml
Following should fail
Successfully trapped invalid schema
Schematron validation
Matching Items: 6
Matching Items: 6
RNG Validation
